.local/matrix-runner-3.sh

```
@@ -0,0 +1,332 @@
+}
+
+# Background sampler — writes 1 row every 5s to $1 CSV.
+start_sampler() {
+  local csv="$1"
+  local stopfile="$csv.stop"
+  rm -f "$stopfile"
+  (
+    echo "ts,conntrack_count,sock_used,sock_tw,sock_orphan,redis_ops_s,redis_cpu_user,redis_cpu_sys,redis_conn_recv,pg_xact_commit,pg_tup_inserted,pg_blks_read,api_access_lines,pem_cpu_m,pem_mem_mi,k6sa_cpu_m,k6sa_mem_mi,coredns_q_total,coredns_cache_miss" > "$csv"
+    # Resolve coredns pod IP once
+    local coredns_ip=$(kubectl get pods -n kube-system -l k8s-app=kube-dns -o jsonpath='{.items[0].status.podIP}' 2>/dev/null)
+    local prev_redis_conn=0 prev_pg_xact=0 prev_pg_tup=0 prev_pg_blks=0 prev_api_lines=0
```

Shellcheck (0.11.0)

[warning] prev_redis_conn appears unused. Verify use (or export if used externally).

(SC2034)